lib/oauth2_basic_authenticator.rb

```diff
@@ -84,7 +84,7 @@ def walk_path(fragment, segments, seg_index = 0)
     return nil unless fragment.is_a?(Hash) || fragment.is_a?(Array)
     first_seg = segments[seg_index].scan(/([\d+])/).length > 0 ? first_seg.split("[")[0] : first_seg
     if fragment.is_a?(Hash)
-      deref = fragment[first_seg] || fragment[first_seg.to_sym]
+      deref = fragment.key?(first_seg) ? fragment[first_seg] : fragment[first_seg.to_sym]
```

Comment:
Can the key ever be a symbol? If the object comes only from `JSON.parse` - I don't think so.

(it was implemented this way since [the first commit](https://github.com/discourse/discourse-oauth2-basic/blame/6260d0e9b0b89054a24cb5730be6dd6b90380cd7/plugin.rb) but that might have been either a defensive coding or specs artificially passed in symbols)

```suggestion
      deref = fragment[first_seg]
```

Reply:
> Can the key ever be a symbol? If the object comes only from JSON.parse - I don't think so.

You sound right. I think you're right. Let's do this.